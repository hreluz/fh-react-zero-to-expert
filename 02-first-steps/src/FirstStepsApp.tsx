import { ItemCounter } from "./shopping-cart/ItemCounter";

export function FirstStepsApp () {
    return (
        <>
            <h1>Shopping Cart</h1>
            <ItemCounter name="Switch 2" quantity={30}/>
            <ItemCounter name="Super Switch"/>
        </>
    )
}