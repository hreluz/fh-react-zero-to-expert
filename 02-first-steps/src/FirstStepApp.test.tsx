import { describe, expect, test } from "vitest";
import { FirstStepsApp } from "./FirstStepsApp";
import { render, screen } from "@testing-library/react";

describe('FirstStepApp', () => {
    test('should match snapshot', () => {
        const {container} = render(<FirstStepsApp/>);

        expect(container).toMatchSnapshot();

        screen.debug()
    });
});