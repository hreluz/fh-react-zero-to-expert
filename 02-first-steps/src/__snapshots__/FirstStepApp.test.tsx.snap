// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FirstStepApp > should match snapshot 1`] = `
<div>
  <h1>
    Shopping Cart
  </h1>
  <section
    class="item-row"
  >
    <span
      class="item-text"
      style="color: black;"
    >
      Nintendo Switch 2
    </span>
    <button>
      +1
    </button>
    <span>
      5
    </span>
    <button>
      -1
    </button>
  </section>
  <section
    class="item-row"
  >
    <span
      class="item-text"
      style="color: black;"
    >
      Play Station 4
    </span>
    <button>
      +1
    </button>
    <span>
      6
    </span>
    <button>
      -1
    </button>
  </section>
  <section
    class="item-row"
  >
    <span
      class="item-text"
      style="color: black;"
    >
      Play Station 5
    </span>
    <button>
      +1
    </button>
    <span>
      3
    </span>
    <button>
      -1
    </button>
  </section>
  <section
    class="item-row"
  >
    <span
      class="item-text"
      style="color: black;"
    >
      Play Station 3
    </span>
    <button>
      +1
    </button>
    <span>
      5
    </span>
    <button>
      -1
    </button>
  </section>
</div>
`;
